### PR TITLE
[Testing] Migration from Cypress to Playwright

### LINKED ISSUE #399: [Testing] Migrate from Cypress to Playwright
- [ ] E2E Testing 
- [ ] Component testing
- [ ] Screenshots
- [ ] Readme update
- [ ] Readme tech-stack logo

### PR DESCRIPTION
**Pull request by DorijanH**
## Added
+ Playwright's E2E and component testing
+ [pixelmatch](https://github.com/mapbox/pixelmatch) for comparing screenshots
+ `dev-open` package.json script

## Changed
+ README documentation
+ README tech-stack logo
+ Screenshots compare functionality and scripts
+ Feature:remove functionality to remove Playwright
+ `BuildAndTest` workflow

## Removed
+ Cypress
+ `ScreenshotsCheck` workflow (integrated inside the BuildAndTest)
+ Unnecessary `package.json` scripts

Closes #399

### CODE REVIEW

**Review by AleksandarDev — APPROVED** [review]

File: .github/workflows/BuildAndTest.yml [hunk 1]
```diff
@@ -36,21 +49,41 @@ jobs:
...
+      # Running application tests
+      - name: Yarn test
+        run: yarn test
+
+      # Running E2E screenshots check
+      - name: E2E check
+        run: yarn e2e-check
+
+      # Committing E2E screenshot changes
+      - name: Commit E2E changes
```
Comment (AleksandarDev) [inline .github/workflows/BuildAndTest.yml:70]: We need permissions flag at the top to do this

File: .github/workflows/BuildAndTest.yml [hunk 2]
```diff
@@ -1,33 +1,46 @@
...
+env:
+  BRANCH_NAME: ${{ github.head_ref || github.ref_name }}
 
+# Triggering the workflow on PRs and pushes to relevant branches
 on:
   pull_request:
     branches:
       - main
+      - feature/**
   push:
```
Comment (AleksandarDev) [inline .github/workflows/BuildAndTest.yml:13]: We don't need to run this on main (once merged)

**Review by AleksandarDev — CHANGES_REQUESTED** [review]

File: .github/workflows/BuildAndTest.yml (hunk 2 above)
Comment (DorijanH) [reply .github/workflows/BuildAndTest.yml]: Removed on `push` event

File: .github/workflows/BuildAndTest.yml (hunk 1 above)
Comment (DorijanH) [reply .github/workflows/BuildAndTest.yml]: Added write `permission:contents` to the job. That was enough to get it working on my test repo

**Review by AleksandarDev — APPROVED** [review]

**Pull request merged**